win32/win32.c

```diff
@@ -7337,8 +7308,7 @@ rb_w32_read_internal(int fd, void *buf, size_t size, rb_off_t *offset)
     ret += read;
     if (read >= len) {
         buf = (char *)buf + read;
-        if (err != ERROR_OPERATION_ABORTED &&
-            !(isconsole && len == 1 && (!islineinput || *((char *)buf - 1) == '\n')) && size > 0)
+        if (err != ERROR_OPERATION_ABORTED && size > 0)
```

Comment:
I'm not sure if this condition is correct (including whether it was originally correct), but I guess it's fine if it works.